### ISSUE TITLE
Deduplicate - lychrel numbers

### ISSUE THREAD

**Issue by hobovsky** [codewars/content-issues#52]
**lychrel number**

Creating tests for this problem is tricky (but not difficult), because some seemingly small inputs can cause either infinite sequence (for example, 196), or can lead to numbers which overflow native numeric types.

1. [Lychrel numbers](https://www.codewars.com/kata/58d4634b14286c0302000593)
  *  JS only, Node 8
  * 7kyu
  * input up to 10600 (most probably chosen so none of values in the input range cause overflow, but have to check), a couple of fixed tests for lychrel numbers
  * has a parameter to stop iterations after some repetitions, what allows to handle potentially infinite sequences
  * 1 (minor) issue reported
  * created Mar 2017, author inactive
  * 93 solutions, 79% satisfaction
  * returned value is number of steps
2. [The 196-algorithm and Lychrel numbers](https://www.codewars.com/kata/55e079b1e00f75e1cd00013b)
  * JS only, Node 8
  * 6kyu
  * input up to 10k
  * explicit tests for long iterations (the longest iteration in the input range is 261)
  * no active issues
  * created Aug 2015, author seems to be active
  * 57 solutions, 89% satisfaction
  * return value is the final palindrome from a generated sequence
3. [Palindromic Numbers](https://www.codewars.com/kata/52a0f488852a85c723000aca)
  * 3 languages (Node10 for JS)
  * 6 kyu
  * only a small set of fixed tests in JS, so no risk of overflow or infinite sequence. No random tests in JS. Ruby has full-range tests up to 5k, and Haskell has some QuickCheck magick which no one knows how it works.
  * 2 reported issues
  * created Dec 2013, autor inactive
  * 572 solutions, 86% satisfaction
  * asks for both: final palindrome and number of steps, but has poor output format

**Comment by hobovsky** [comment]
My proposal:

Retire all of them, and replace with a new kata which would go through regular beta process. This would let users decide on its rank. I don't think any of existing kata is worth keeping, because each of them poses some maintenance effort.


If you prefer to keep one of existing kata, then I have no good candidate - I am ok with each of them, because I do not like any of them too much :)

**Comment by ejini6969** [comment]
The first and second one have to go as the description is way too long and has formatting issues, plus JS version update required. And providing a threshold to number of iterations before halting which allow multiple different solutions to bypass is not a good idea IMO. 

The last one can be kept as it is the combination of previous two plus it guarantees that inputs will be palindromic numbers and not `lychrel numbers`, its only "not so good" part is the output format, should be a tuple or list instead depending on languages.

I vote to retire first two and send the last one back to beta for reranking (if required) and issue fixing.

**Comment by hobovsky** [comment]
Oh I didn't think about the possibility of sending back to beta, thanks!

**Comment by CiprianAmza** [comment]
Nice job, hobovsky, for finding such similar katas. The last kata is the combination of the first two and requires both the total steps needed to reach the number and the number itself, while the first requires only the steps and the second requires just the number itself. Thinking of this, I assume that the last kata should be kept. Tests were pretty bad for all the three katas (The second kata was easy to brute force, as all the big numbers expected a -1). The 7 kyu kata tests were pretty bad (sometimes the same input required different output). In conclusion, I would suggest to keep only the `Palindromic Numbers` kata and to improve the tests, because it is better written, more complex and has translations.

**Comment by Kacarott** [comment]
I disagree with ejini. I think the main interesting factor of these katas is that there exist numbers which somehow never fall into palindromes (the `lychrel numbers`) although, simply setting an arbitrary limit does feel a little clunky.

I would vote to keep the second one, especially as its author is the only somewhat active one. We could send it back to beta, raise a bunch of issues to fix it up, and rerank it. The first one required an extra parameter which felt even stranger than a hardcoded limit (why would we want to search further for some numbers, than for others?), and the third one wants us to return two numbers, *as a string*, and also just avoids computing lychrel numbers, encouraging solutions which will break when given a simple int (and also making it less interesting, as said above).

I also think that the third task is *not* a combination of the first two, it is exactly the same task, with trivial changes to the output value. Any solution which finds the palindrome can trivially count the steps, and any solution which counts steps will already have the palindrome. And asking for *both* of these values feels unnecessary, especially when done as a string.

**Comment by hobovsky** [comment]
> Nice job, hobovsky, for finding such similar katas.

To be fair to everyone involved, I do not really find these kata, i take them from the list posted by ejini in #39, and the list was either created by him, or takem from the old CW wiki where it was created collectively - we'd have to ask ejini to know for sure.

All I do is just trying to coordinate the efforts here and get things moving.

**Comment by ejini6969** [comment]
> To be fair to everyone involved, I do not really find these kata, i take them from the list posted by ejini in #39, and the list was either created by him, or takem from the old CW wiki where it was created collectively - we'd have to ask ejini to know for sure.

I mainly gathered all these katas based on kata discourse discussion and some recall from previous solving experience which can be easy or hard based on the kata's requirements.

> I disagree with ejini. I think the main interesting factor of these katas is that there exist numbers which somehow never fall into palindromes (the lychrel numbers) although, simply setting an arbitrary limit does feel a little clunky.

Okay, you got your point there too, we shall see how this goes at the end of the day !!!

**Comment by JohanWiltink** [comment]
>     3. [Palindromic Numbers](https://www.codewars.com/kata/52a0f488852a85c723000aca)
> 
>     * .. Haskell has some QuickCheck magick which no one knows how it works.

QuickCheck without size modifiers generates numbers in `[0..100]`. These are then filtered for non-Lychrel-numbers ( in `1 000` iterations ) only.
